File: DuktapeDemo/DuktapeEngineDemo/src/main/java/com/furture/react/activity/test/TestRef.java
package com.furture.react.activity.test;

import android.util.Log;

import com.furture.react.JSRef;

public class TestRef {
	JSRef jsRef;
	public TestRef(JSRef jsRef){
		this.jsRef = jsRef;
	}
	
	public Object callJsRef(JSRef args){
		if(jsRef != args){
			throw new RuntimeException("TestRef Failed Must Same ref");
		}
		Log.e("ScriptEngine", "ScriptEngine JSRef " + args.getRef());
		return jsRef;
	}


	public Object javaRef(){
		Log.e("ScriptEngine", "ScriptEngine  javaRef ");
		return this;
	}
}

File: DuktapeDemo/DuktapeEngineDemo/src/main/assets/testRef.js
importClass("com.furture.react.activity.test.TestRef")


var a = {say:function(){
	  print("save hello world");
}};

var testRef = new TestRef(a);

var b = testRef.callJsRef(a);

for(var i=0; i<10; i++){
	testRef.callJsRef(a);
	testRef.callJsRef(b);
}

b.say();
a.say();
print(a == b)

for(var i=0; i< 10; i++){
     testRef.javaRef();
}


true;